adapters/rise/rise.go

```diff
@@ -0,0 +1,134 @@
+package rise
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/prebid/openrtb/v19/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+// adapter is a Rise implementation of the adapters.Bidder interface.
+type adapter struct {
+	endpointURL string
+}
+
+func Builder(_ openrtb_ext.BidderName, config config.Adapter, _ config.Server) (adapters.Bidder, error) {
+	return &adapter{
+		endpointURL: config.Endpoint,
+	}, nil
+}
+
+// MakeRequests prepares the HTTP requests which should be made to fetch bids.
+func (a *adapter) MakeRequests(
+	openRTBRequest *openrtb2.BidRequest,
+	_ *adapters.ExtraRequestInfo) (
+	requestsToBidder []*adapters.RequestData,
+	errs []error,
+) {
+	publisherID, err := a.extractPublisherID(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("extractPublisherID: %w", err))
+		return nil, errs
+	}
+
+	openRTBRequestJSON, err := json.Marshal(openRTBRequest)
+	if err != nil {
+		errs = append(errs, fmt.Errorf("marshal bidRequest: %w", err))
+		return nil, errs
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+
+	return append(requestsToBidder, &adapters.RequestData{
+		Method:  http.MethodPost,
+		Uri:     a.endpointURL + "?publisher_id=" + publisherID,
+		Body:    openRTBRequestJSON,
+		Headers: headers,
+	}), nil
+}
+
+func (a *adapter) extractPublisherID(openRTBRequest *openrtb2.BidRequest) (string, error) {
+	var (
+		bidderExt   adapters.ExtImpBidder
+		impExt      openrtb_ext.ImpExtRise
+		publisherID string
+	)
+
+	for _, imp := range openRTBRequest.Imp {
+		if err := json.Unmarshal(imp.Ext, &bidderExt); err == nil {
+			if err = json.Unmarshal(bidderExt.Bidder, &impExt); err == nil && impExt.PublisherID != "" {
+				publisherID = strings.TrimSpace(impExt.PublisherID)
+				break
+			}
+		}
+	}
+	if publisherID == "" {
+		return "", errors.New("no publisherID supplied")
+	}
+
+	return publisherID, nil
+}
+
+// MakeBids unpacks the server's response into Bids.
+func (a *adapter) MakeBids(
+	request *openrtb2.BidRequest,
+	_ *adapters.RequestData,
+	responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
```

Comment:
Checking response status code this way is correct but you could also make use of the following code - 
```
if httputil.IsResponseStatusCodeNoContent(response) {
		return nil, nil
	}

	if err := httputil.CheckResponseStatusCodeForErrors(response); err != nil {
		return nil, []error{err}
	}
```